src/dependencies.rb

```diff
@@ -0,0 +1,110 @@
+#==============================================================================
+# Copyright (C) 2017 Stephen F. Norledge and Alces Software Ltd.
+#
+# This file/package is part of Alces Metalware.
+#
+# Alces Metalware is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation, either version 3 of
+# the License, or (at your option) any later version.
+#
+# Alces Metalware is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this package.  If not, see <http://www.gnu.org/licenses/>.
+#
+# For more information on the Alces Metalware, please visit:
+# https://github.com/alces-software/metalware
+#==============================================================================
+require 'exceptions'
+require 'constants'
+
+module Metalware
+  class Dependencies
+    def initialize(metal_config, command_input, dep_hash = {})
+      @config = metal_config
+      @dep_hash = dep_hash
+      @command = command_input
+    end
+
+    def enforce
+      @dep_hash.each { |dep, values|
+        next unless values # Rejects nil and false
+        values = [values] unless values.is_a?(Array)
+        values.each { |value| validate_dependency(dep, value) }
+      }
+    end
+
+    private
+
+    attr_reader :config, :command
+
+    def validate_dependency(dep, value)
+      return if send(:"validate_#{dep}") == "skip"
+      return if value.is_a?(TrueClass)
+      unless file_exists?(dep, value)
+        raise DependencyFailure, get_generic_failure_message(dep, value)
+      end
+    end
+
+    def validate_repo
+      @validated_repo ||= begin
+        msg = "'#{command}' requires a repo. Please run 'metal repo use'"
+        raise DependencyFailure, msg unless file_exists?(:repo, "", true)
+        true
```

Comment:
The purposes of the true is to set a value for `@validate_repo`. Originally this was called multiple times so it allows for the method to be short circuited as it only needs to be checked once. Now that `Dependencies` has been refactored, it doesn't get called as many times. However it is still will be called more than once. 

I have added a comment explaining this in the new version